src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8131,6 +8131,29 @@ public static string GetTarget(PSObject instance)
             return null;
         }
 
+        /// <summary>
+        /// Gets the target for a given file or directory, resolving symbolic links.
+        /// </summary>
+        /// <param name="instance">The FileInfo or DirectoryInfo type.</param>
+        /// <returns>The file path the instance points to.</returns>
+        public static FileSystemInfo ResolvedTarget(PSObject instance)
```

Comment:
```suggestion
        public static string ResolvedTarget(PSObject instance)
```

Reply:
I pushed a change that should resolve this. However, with the code suggested earlier, we return the path to the destination of a broken symbolic link. Should this remain the case? Thanks!

Reply:
I agree to follow .Net behavior.

Reply:
@hammy3502 about your question, please take a look at https://github.com/PowerShell/PowerShell/pull/16490#discussion_r782800772